src/Core/System/SystemConfig/Command/ConfigSet.php

```diff
@@ -30,17 +30,31 @@ protected function configure(): void
             ->addArgument('key', InputArgument::REQUIRED)
             ->addArgument('value', InputArgument::REQUIRED)
             ->addOption('salesChannelId', 's', InputOption::VALUE_OPTIONAL)
-        ;
+            ->addOption('decode', 'd', InputOption::VALUE_NONE, 'If provided, the input value will be interpreted as JSON. Use this option to provide values as boolean, integer or float.');
     }
 
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
         $this->systemConfigService->set(
             $input->getArgument('key'),
-            $input->getArgument('value'),
+            $this->handleDecode($input),
             $input->getOption('salesChannelId')
         );
 
         return 0;
     }
+
+    protected function handleDecode(InputInterface $input)
+    {
+        $value = $input->getArgument('value');
+        if ($input->getOption('decode')) {
+            $decodedValue = json_decode($value, true);
```

Comment:
```suggestion
            return json_decode($value, true, 512, \JSON_THROW_ON_ERROR);
```
by using the Throw on error flag you can get rid of the manual check to the json erros

and besides that i would prefer throwing an exception in the case that the decode option is used, but the provided value can't be decoded instead of silencing the error and continuing with the raw value

Can you also add a test case for invalid json?

Reply:
if i use the `JSON_THROW_ON_ERROR` decoding a simple string will throw error so the test case
`yield 'Decode string remains string' => ['random string', 'random string', true];` will not work anymore since it will throw error.